packages/paste-core/components/editable-code-block/src/EditableCodeBlock.tsx

```diff
@@ -6,34 +6,149 @@ import {
+
+const LoadingMessage: React.FC<{i18nLoadingLabel: string}> = ({i18nLoadingLabel}) => (
+  <>
+    <Spinner color="colorTextPrimaryWeak" decorative />
+    <ScreenReaderOnly>{i18nLoadingLabel}</ScreenReaderOnly>
+  </>
+);
+LoadingMessage.displayName = 'LoadingMessage';
 
 export interface EditableCodeBlockProps
-  extends Omit<CodeEditorProps, 'wrapperProps' | 'className' | 'loading' | 'theme'> {
+  extends Omit<CodeEditorProps, 'wrapperProps' | 'className' | 'loading' | 'theme' | 'options'> {
+  children?: never;
+  /*
+   * NOTE: I leave these commented out because I'm not sure whether we want to allow users to override these options.
+   * If we do, we can uncomment these easily in the future since the work is already done.
+   */
+  // monacoOptions?: CodeEditorProps['options'];
   element?: BoxProps['element'];
+  minimap?: boolean;
+  lineNumbers?: 'on' | 'off';
+  folding?: boolean;
+  readOnly?: boolean;
+  indentationGuide?: boolean;
+  showMinimap?: boolean;
+  scrollBeyondLastLine?: boolean;
+  inlineErrorRange?: IRange;
+  inlineErrorHoverMessage?: IMarkdownString | IMarkdownString[] | null;
+  inlineErrorIsWholeLine?: boolean;
+  i18nLoadingLabel?: string;
 }
 
 export const EditableCodeBlock: React.FC<EditableCodeBlockProps> = ({
   onMount,
   element = 'EDITABLE_CODE_BLOCK',
+  // monacoOptions = {},
+  lineNumbers = 'on',
+  readOnly = false,
+  folding = true,
+  indentationGuide = true,
+  showMinimap = false,
+  scrollBeyondLastLine = false,
+  inlineErrorRange,
+  inlineErrorHoverMessage,
+  inlineErrorIsWholeLine = true,
+  i18nLoadingLabel = 'Loading code...',
   ...props
 }) => {
-  const handleEditorDidMount = (editor: Editor.IStandaloneCodeEditor, monaco: Monaco): void => {
-    // Sets the Paste theme for the editor
-    monaco.editor.defineTheme('paste', CodeEditorPasteTheme);
-    monaco.editor.setTheme('paste');
-
-    /*
-     * Call provided onMount function.
-     * This can be used to add custom language support, or to grab a
-     * reference to the ref (monaco) object.
-     */
-    onMount?.(editor, monaco);
-  };
+  const controlledOptions = React.useMemo(() => {
+    return {
+      lineNumbers,
+      folding,
+      readOnly,
+      scrollBeyondLastLine,
+      glyphMargin: inlineErrorRange != null,
+      // ...monacoOptions,
+      minimap: {
+        enabled: showMinimap,
+        // ...monacoOptions.minimap,
+      },
+      guides: {
+        indentation: indentationGuide,
+        // ...monacoOptions.guides,
+      },
+    } as CodeEditorProps['options'];
+  }, [
+    // monacoOptions,
+    readOnly,
+    showMinimap,
+    folding,
+    inlineErrorRange,
+    scrollBeyondLastLine,
+    indentationGuide,
+    lineNumbers,
+  ]);
+
+  const handleEditorDidMount = React.useCallback(
+    (editor: Editor.IStandaloneCodeEditor, monaco: Monaco): void => {
+      // Sets the Paste theme for the editor
+      monaco.editor.defineTheme('paste', CodeEditorPasteTheme);
```

Comment:
issue: can you modify the theme to set the font-family to twilio sans mono please?

Reply:
comment: The error dialog text is also using the system font. There's no way that it can change typeface based on theme, right?

Alternate suggestion is to set it to Inter, but I don't feel like that's worth it since it's so close to looking like a system font anyway.

Reply:
Done